lib/quadquizaminos/bottom.ex

```diff
@@ -24,13 +31,16 @@ defmodule Quadquizaminos.Bottom do
   end
 
   def complete?(bottom, row) do
+    ignore_result = vulnerable?(bottom, row)
+
     count =
       bottom
       |> Map.keys()
       |> Enum.filter(fn {_x, y} -> y == row end)
       |> Enum.count()
 
-    count == 10
+    complete_row? = count == 10
+    complete_row? and !ignore_result 
```

Comment:
Suggestion: What about if we leave complete?/2 to only check for complete rows and then check for both complete rows and vulnerability at this point

```suggestion
    def complete_ys(bottom) do
    bottom
    |> Map.keys()
    |> Enum.map(&elem(&1, 1))
    |> Enum.uniq()
    |> Enum.filter(fn row -> complete?(bottom, row) and ! has_vulnerable_block(bottom, row) end)
  end

```